src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -373,7 +373,10 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
                 return null;
             }
 
-            $collectionValueType = $type->isCollection() ? $type->getCollectionValueType() : null;
+            $collectionValueType = $type->isCollection() ? $type->getCollectionValueTypes() : null;
+            if (null !== $collectionValueType) {
+                [$collectionValueType] = $collectionValueType;
```

Comment:
What if `getCollectionValueTypes` returns an empty array (because Type were initialized with an empty array)?

I'd rather removes nullables values (always return array) and use `$collectionValueType = $type->isCollection() ? $type->getCollectionValueTypes()[0] ?? null : null;